src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -411,7 +413,7 @@ private bool StartCore(ProcessStartInfo startInfo)
                 {
                     argv = ParseArgv(startInfo);
 
-                    isExecuting = ForkAndExecProcess(filename, argv, envp, cwd,
+                    isExecuting = ForkAndExecProcess(originalFilename, filename, argv, envp, cwd,
```

Comment:
Because `filename` gets modified by the code before calling ForkAndExecProcess and might be null/empty. I thought it would be better to use the filename from startInfo in such cases instead of using an empty filename in the exception text.
originalFilename is only used if filename is null/empty.

Reply:
I see, thanks. In what situation might it go from non-empty/non-null to empty/null?

Reply:
`ResolveExecutableForShellExecute` and `ResolvePath` try to find the given file in various locations and return `null` the file does not exist in any of those locations.
That's of course totally different than the windows implementation where the filename gets passed to `CreateProcess` and that API is responsible for resolving the path.
So the exception thrown on windows will always contain the original filename.